sdk/src/client/node_api/indexer/query_parameters.rs

```diff
@@ -3,343 +3,295 @@
-            Self::MaxNativeTokenCount(_) => 14,
-            Self::MinNativeTokenCount(_) => 15,
-            Self::PageSize(_) => 16,
-            Self::Sender(_) => 17,
-            Self::StateController(_) => 18,
-            Self::StorageDepositReturnAddress(_) => 19,
-            Self::Tag(_) => 20,
-            Self::TimelockedAfter(_) => 21,
-            Self::TimelockedBefore(_) => 22,
-            Self::UnlockableByAddress(_) => 23,
-        }
-    }
-}
-
-impl fmt::Display for QueryParameter {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let query_tuple = self.to_query_tuple();
-        write!(f, "{}={}", query_tuple.0, query_tuple.1)
+impl BasicOutputQueryParameters {
+    /// Sets `.address(address).has_expiration(false).has_storage_deposit_return(false).has_timelock(false)` to only
+    /// get outputs that can be unlocked by the address without potential further restrictions.
+    pub fn only_address_unlock_condition(self, address: impl Into<Bech32Address>) -> Self {
+        self.address(address.into())
+            .has_expiration(false)
+            .has_storage_deposit_return(false)
+            .has_timelock(false)
     }
 }
 
-macro_rules! verify_query_parameters {
-    ($query_parameters:ident, $first:path $(, $rest:path)*) => {
-        if let Some(qp) = $query_parameters.iter().find(|qp| {
-            !matches!(qp, $first(_) $(| $rest(_))*)
-        }) {
-            Err(Error::UnsupportedQueryParameter(qp.clone()))
-        } else {
-            Ok(())
-        }
-    };
+/// Query parameter for output requests.
+#[derive(Setters, Debug, Default, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[setters(strip_option)]
+#[serde(rename_all = "camelCase")]
+pub struct AccountOutputQueryParameters {
+    /// Returns outputs that were created after a certain slot index.
+    pub created_after: Option<SlotIndex>,
+    /// Returns outputs that were created before a certain slot index.
+    pub created_before: Option<SlotIndex>,
+    /// Starts the search from the cursor (confirmationMS+outputId.pageSize).
+    pub cursor: Option<String>,
+    /// The maximum amount of items returned in one call. If there are more items, a cursor to the next page is
+    /// returned too. The parameter is ignored when pageSize is defined via the cursor parameter.
+    pub page_size: Option<usize>,
+    /// Returns outputs that are unlockable by the bech32 address.
+    pub unlockable_by_address: Option<Bech32Address>,
+    /// Filters outputs based on the presence of validated Sender (bech32 encoded).
+    pub sender: Option<Bech32Address>,
+    /// Filters outputs based on bech32-encoded issuer address.
+    pub issuer: Option<Bech32Address>,
+    /// Filters outputs based on bech32-encoded governor (governance controller) address.
+    pub governor: Option<Bech32Address>,
+    /// Filters outputs based on bech32-encoded state controller address.
+    pub state_controller: Option<Bech32Address>,
```

Comment:
I would make all fields private. We only need to serialise the struct, no need to access the fields. Derive getters if needed. But as we've seen with Alex's PR, public fields like that are shooting ourselves in the foot as adding more fields becomes breaking change.

Reply:
Changed